src/Microsoft.Azure.WebJobs.Host/Executors/FunctionInvoker.cs

```diff
@@ -48,6 +48,9 @@ public IReadOnlyList<string> ParameterNames
 
         public async Task InvokeAsync(object[] arguments)
         {
+            // Return a task immediately in case the method is not async.
+            await Task.Yield();
```

Comment:
We don't get a Task returned to us until the first await -- in this case it's  _methodInvoker.InvokeAsync() below.

If the function ends up being synchronous and has an infinite loop, we'll never actually return a Task. That means we can't cancel because we're effectively blocked here.

---
In reply to: [77048332](https://github.com/Azure/azure-webjobs-sdk/pull/799#discussion_r77048332) [](ancestors = 77048332)